platform-jobs/samza/publish-pipeline/src/main/java/org/ekstep/jobs/samza/service/PublishPipelineService.java

```diff
@@ -252,5 +257,52 @@ private boolean validateObject(Map<String, Object> edata) {
                 return true;
         }
         return false;
-    }	
+    }
+
+	private void pushInstructionEvent(Node node) throws Exception{
+		Map<String,Object> actor = new HashMap<String,Object>();
+		Map<String,Object> context = new HashMap<String,Object>();
+		Map<String,Object> object = new HashMap<String,Object>();
+		Map<String,Object> edata = new HashMap<String,Object>();
+
+		generateInstructionEventMetadata(actor, context, object, edata, node.getMetadata(), node.getIdentifier());
+		String beJobRequestEvent = LogTelemetryEventUtil.logInstructionEvent(actor, context, object, edata);
+		String topic = Platform.config.hasPath("post_publish_event_topic") ? Platform.config.getString("post_publish_event_topic"): "post.publish.request";
+		if(StringUtils.isBlank(beJobRequestEvent)) {
+			TelemetryManager.error("Post Publish event is not generated properly. #postPublishJob : " + beJobRequestEvent);
+			throw new ClientException("BE_JOB_REQUEST_EXCEPTION", "Event is not generated properly.");
+		}
+		if(StringUtils.isNotBlank(topic)) {
```

Comment:
Instead of calling Kafka client, we can directly push event via collector
collector.send(new OutgoingMessageEnvelope(new SystemStream("kafka", <topicName>, event <Map>));
Refer AbstractTask class: 
```@Override
	public void window(MessageCollector collector, TaskCoordinator coordinator) throws Exception {
		Map<String, Object> event = metrics.collect();
		collector.send(new OutgoingMessageEnvelope(new SystemStream("kafka", metrics.getTopic()), event));
		metrics.clear();
	}```

Reply:
Just pass `collector` instance to pushInstructionEvent method.

Reply:
Handle exception also

Reply:
Done